crates/forge/src/new.rs

```diff
@@ -48,11 +48,21 @@ struct TemplateManifestConfig {
 }
 
 impl TemplateManifestConfig {
-    fn add_dependencies(&self, scarb_manifest_path: &PathBuf) -> Result<()> {
+    fn add_dependencies(
+        &self,
+        scarb_version: &Version,
+        scarb_manifest_path: &PathBuf,
+    ) -> Result<()> {
         if env::var("DEV_DISABLE_SNFORGE_STD_DEPENDENCY").is_err() {
+            // TODO: Check Scarb `2.12`
+            let snforge_package = if scarb_version.build.is_empty() {
+                "snforge_std_compatibility"
+            } else {
+                "snforge_std"
+            };
```

Comment:
nit: This logic could be moved to separate method, e.g. `resolve_snforge_std_package`